app/controllers/quizzes_controller.rb

```diff
@@ -28,15 +28,56 @@ def new
 
   # POST /quizzes
   def create
-    @quiz = current_user.quizzes.build(quiz_params)  # Associate the quiz with the current_user
-    # service = OpenAIService.new
-    # prompt = params[:prompt]
-    # @response = service.generate_response(prompt)
-    if @quiz.save
-      redirect_to quizzes_path, notice: "Quiz was successfully generated."
-    else
-      flash[:alert] = "Quiz could not be created"
-      render :new, status: :unprocessable_entity
+    prompt = <<~PROMPT
+      Generate a multiple-choice questions (MCQs) based quiz using the following inputs:
+      Topic (of the quiz): #{params[:quiz][:topic]}.
+      Number of Questions (in the quiz): #{params[:quiz][:number_of_questions]}.
+      Difficulty (of the quiz): #{params[:quiz][:difficulty]}.
+      Detail Level (depth of the questions): #{params[:quiz][:detail_level]}.
+      Instructions:
+      For each question, generate a question and its correct answer (appropriate to the input difficulty and detail level).
+      2. Create three plausible but incorrect options (distractors) for each question.
+      3. Return only a valid JSON object. Do not include any text before or after the JSON. The JSON should strictly follow this structure:
+
+      {
+        "questions": [
+          {
+            "content": "Question text",
+            "options": ["Option 1", "Option 2", "Option 3", "Option 4"],
+            "correct_answer": "Option 1"
+          },
+          ...
+        ]
+      }
+
+    Do not deviate from this format. Do not include extra characters or quotes outside the JSON structure.
+    PROMPT
+    begin
+      openai_service = OpenaiService.new
+      response = openai_service.generate_response(prompt, 10000, "mixtral-8x7b-32768")
+      session[:response] = nil
```

Comment:
Here, I set the tokens to 10000 because I found that having >15 questions with 32k tokens makes the JSON parsed incorrectly. Also we don't want to waste tokens for rate limiting purpose.

Reply:
This makes sense. Perhaps we should then also add an upper bound on the number_of_questions that can be specified through the form in views/quizzes/new.html.erb ?

Reply:
The upper bound right now is 20 which I think is reasonable.